WordPress/src/main/java/org/wordpress/android/ui/accounts/login/LoginGoogleFragment.java

```diff
@@ -0,0 +1,327 @@
+import com.google.android.gms.common.api.GoogleApiClient;
+import com.google.android.gms.common.api.GoogleApiClient.ConnectionCallbacks;
+import com.google.android.gms.common.api.GoogleApiClient.OnConnectionFailedListener;
+
+import org.greenrobot.eventbus.Subscribe;
+import org.greenrobot.eventbus.ThreadMode;
+import org.wordpress.android.R;
+import org.wordpress.android.WordPress;
+import org.wordpress.android.analytics.AnalyticsTracker;
+import org.wordpress.android.fluxc.Dispatcher;
+import org.wordpress.android.fluxc.generated.AccountActionBuilder;
+import org.wordpress.android.fluxc.store.AccountStore.OnAuthenticationChanged;
+import org.wordpress.android.fluxc.store.AccountStore.OnSocialChanged;
+import org.wordpress.android.fluxc.store.AccountStore.PushSocialLoginPayload;
+import org.wordpress.android.util.AppLog;
+import org.wordpress.android.util.AppLog.T;
+
+import javax.inject.Inject;
+
+import static android.app.Activity.RESULT_OK;
+
+public class LoginGoogleFragment extends Fragment implements ConnectionCallbacks, OnConnectionFailedListener {
+    private GoogleApiClient mGoogleApiClient;
+    private LoginListener mLoginListener;
+    private String mGoogleEmail;
+    private String mIdToken;
+    private boolean isResolvingError;
+    private boolean shouldResolveError;
+
+    private static final String SERVICE_TYPE_GOOGLE = "google";
+    private static final String STATE_RESOLVING_ERROR = "STATE_RESOLVING_ERROR";
+    private static final int REQUEST_CONNECT = 1000;
+    private static final int REQUEST_LOGIN = 1001;
+
+    public static final String TAG = "login_google_fragment_tag";
+
+    @Inject Dispatcher mDispatcher;
+
+    @Override
+    public void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        ((WordPress) getActivity().getApplication()).component().inject(this);
+
+        // Restore state of error resolving.
+        isResolvingError = savedInstanceState != null && savedInstanceState.getBoolean(STATE_RESOLVING_ERROR, false);
+
+        // Configure sign-in to request user's ID, basic profile, email address, and ID token.
+        // ID and basic profile are included in DEFAULT_SIGN_IN.
+        GoogleSignInOptions googleSignInOptions = new GoogleSignInOptions.Builder(GoogleSignInOptions.DEFAULT_SIGN_IN)
+                .requestServerAuthCode(getString(R.string.default_web_client_id))
+                .requestIdToken(getString(R.string.default_web_client_id))
+                .requestProfile()
+                .requestEmail()
+                .build();
+
+        // Build Google API client with access to sign-in API and options specified above.
+        mGoogleApiClient = new GoogleApiClient.Builder(getActivity())
+                .addApi(Auth.GOOGLE_SIGN_IN_API, googleSignInOptions)
+                .addConnectionCallbacks(this)
+                .addOnConnectionFailedListener(this)
+                .build();
+
+        if (!isResolvingError) {
```

Comment:
The `savedInstanceState` value is checked and `isResolvingError` is restored [a few lines above](https://github.com/wordpress-mobile/WordPress-Android/pull/6764/files/029f3323543e5df45de69bb2ba4469fa69eb11f7#diff-a2548661d9da48177d493d57e932ba55R69).

Reply:
Rotating the device while the google signin process hasn't finished would still trigger the `connectGoogleClient()` line since the `isResolvingError` would still be false. So, a check for rotation would be needed.

Thing is, with the new `setRetainInstance(true)` change, `onCreate` is not hit again so this is no longer an issue.